be/src/storage/persistent_index.cpp

```diff
@@ -315,6 +319,20 @@ StatusOr<std::unique_ptr<ImmutableIndexShard>> ImmutableIndexShard::create(size_
     if (kv_refs.size() == 0) {
         return std::make_unique<ImmutableIndexShard>(0);
     }
+    for (size_t npage = npage_hint; npage < page_max; npage++) {
+        auto rs_create = ImmutableIndexShard::try_create(kv_size, kv_refs, npage);
+        // increase npage and retry
+        if (!rs_create.ok()) {
+            continue;
+        }
+        return std::move(rs_create.value());
+    }
+    return Status::InternalError("failed to create immutable index shard");
+}
+
+StatusOr<std::unique_ptr<ImmutableIndexShard>> ImmutableIndexShard::try_create(size_t kv_size,
```

Comment:
try_create -> create

Reply:
> try_create -> create

There is already a `create` function. `create` function will call `try_create` with a fixed npage, if `try_create` failed, it will increase napge and retry